src/main/lrsql/admin/interceptors/openapi.clj

```diff
@@ -0,0 +1,20 @@
+(ns lrsql.admin.interceptors.openapi
+  (:require [com.yetanalytics.gen-openapi.core :as oa-core]
+            [com.yetanalytics.gen-openapi.generate.schema :as gs]
+            [com.yetanalytics.lrs.pedestal.openapi :as lrs-oa]
+            [io.pedestal.interceptor :refer [interceptor]]))
+
+(defn openapi [routes version]
+  (interceptor
+   {:name ::openapi
+    :enter (fn openapi [ctx]
+             (assoc ctx :response
+                    {:status 200
+                     :body (oa-core/make-oa-map
+                            {:openapi "3.0.0"
+                             :info {:title "LRSQL"
+                                    :version version}
+                             :externalDocs {:url "https://github.com/yetanalytics/lrsql/blob/main/doc/endpoints.md"}
+                             :components (gs/dsl (add-lrsql-specifics
+                                                  lrs-oa/components))}
+                            routes)}))}))
```

Comment:
Since you already have this wrapped inside a function, why not do this in a let binding above so it just happens the once on startup?

Reply:
:+1: